### ISSUE TITLE
Breaking older Julia in downstream packages

### ISSUE THREAD

**Issue by MilesCranmer** [JuliaDiff/DifferentiationInterface.jl#597]
It looks like the Project.toml for DifferentiationInterface.jl is still set to `julia = "1.6"` even though the CI only tests 1.10+.

@gdalle could you please update the Project.toml to Julia 1.10 and also backport this to the registry to the last version that you verified against 1.6?
DifferentiationInterface.jl is now breaking the tests of SymbolicRegression.jl and PySR on older versions of Julia.

(People who are not heavily involved in the Julia community take a while longer to update their versions, and they will inevitably hit these issues. So it's important to constrain the Julia version correctly in Project.toml to prevent this.)

**Comment by gdalle** [comment]
OUCH that is a really bad oversight on my end. Will fix it right away. Thank you for reporting!

**Comment by gdalle** [comment]
Ok so support for Julia 1.6 was first dropped in:

- DifferentiationInterface [v0.6.8](https://github.com/gdalle/DifferentiationInterface.jl/releases/tag/DifferentiationInterface-v0.6.8)
- DifferentiationInterfaceTest [v0.8.0](https://github.com/gdalle/DifferentiationInterface.jl/releases/tag/DifferentiationInterfaceTest-v0.8.0)

Once #598 is merged I'll open a PR to General

**Comment by MilesCranmer** [comment]
Many thanks!!

**Comment by gdalle** [comment]
Now we only need the General PR to be reviewed and merged

**Comment by Red-Portal** [comment]
@gdalle  I think this should have been a minor version increment for this, not just a patch version increment? This is a breaking change after all.

**Comment by gdalle** [comment]
It's actually not a breaking change when done properly because users who stay on the old (unsupported) version of Julia will never even see the newer versions of the package: they can't download those versions at all. See [this section](https://github.com/SciML/ColPrac?tab=readme-ov-file#dropping-support-for-earlier-versions-of-julia) of the ColPrac guidelines for more discussion.
Of course here I screwed up badly by dropping support for 1.6 without actually bumping the compat bound, so users of Julia 1.6 can keep downloading versions of DI which are much too recent. This is why I had to make a manual PR to the General Registry and fix those compat bounds _a posteriori_. Everything is ready but the maintainers of General have some new infrastructure they're putting in place, so the merge is slightly delayed.
